src/power.jl

```diff
@@ -447,6 +511,37 @@ for T = (:Taylor1, :TaylorN)
 
             return nothing
         end
+
+        # in-place squaring: given `c`, compute expansion of `c^2` and save back into `c`
+        @inline function sqr!(c::$T{T}, k::Int) where {T<:NumberNotSeries}
```

Comment:
I think the problem in the tests is related to this method. The following function
```julia
function mysquare(x::Taylor1)
     aux = copy(x)
     for k in reverse(eachindex(aux))
         TS.sqr!(aux, k)
     end
     return aux
 end
```
should compute the square of a Taylor1 object, and has the same problem for `(Taylor1(10)-1)^4`.